src/utils/storage.js

```diff
@@ -107,55 +96,28 @@ export function findSimilarInJar (keyLike) {
  * @param {StorageOptions} storageOptions
  * @returns void
  */
-export function setCookie (key, value, storageOptions) {
+function setCookieInt (key, value, storageOptions) {
   cookies.set(key, value, storageOptions)
 }
 
 /**
  * @param {string} key
  * @param {string} value
+ * @param {number} expires
+ * @param {string} sameSite
+ * @param {string} domain
  * @param {StorageOptions} storageOptions
- * @returns {string|null}
- * @private
- */
-function _cookieGetOrAdd (key, value, storageOptions) {
-  let ret = null
-  try {
-    const oldCookie = getCookie(key)
-    if (oldCookie) {
-      setCookie(key, oldCookie, storageOptions)
-    } else {
-      setCookie(key, value, storageOptions)
-    }
-    ret = getCookie(key)
-  } catch (e) {
-    emitter.error('CookieGetOrAdd', 'Failed manipulating cookie jar', e)
-  }
-  return ret
-}
-
-/**
- * @param {string} key
- * @returns {string|null}
- * @private
+ * @returns void
  */
-export function fetchFromLs (key) {
-  let ret = null
-  try {
-    if (hasLocalStorage()) {
-      ret = _unsafeGetFromLs(key)
-    }
-  } catch (e) {
-    emitter.error('LSFetch', 'Failed fething key from ls', e)
-  }
-  return ret
+export function setCookie (key, value, expires, sameSite, domain) {
+  setCookieInt(key, value, { domain: domain, expires: expires, samesite: sameSite })
```

Comment:
what is the purpose of extracting the `setCookieInt` into a separate method?

Reply:
Because the underlying lib expects the JSON with options.

Reply:
it is possible to just call
```
cookies.set(key, value, { domain: domain, expires: expires, samesite: sameSite })
```